src/samples/http/server-sse.c

```diff
@@ -0,0 +1,210 @@
+        }
+    }
+
+    SOL_PTR_VECTOR_FOREACH_IDX (&responses, sse, i)
+        sol_http_response_sse_send_data(sse, &value);
+
+    return true;
+
+err:
+    sol_quit_with_code(EXIT_FAILURE);
+    return false;
+}
+
+static void
+delete_cb(void *data, const struct sol_http_response_sse *sse)
+{
+    sol_ptr_vector_remove(&responses, sse);
+}
+
+static int
+request_cb(void *data, struct sol_http_request *request)
+{
+    int ret;
+    struct sol_http_response_sse *sse;
+
+    sse = sol_http_server_send_response_sse(request, delete_cb, NULL);
+    if (!sse)
+        return -1;
+
+    ret = sol_ptr_vector_append(&responses, sse);
+    if (ret < 0) {
+        sol_http_response_sse_del(sse);
+        return ret;
+    }
+
+    return 0;
+}
+
+static void
+startup_server(void)
+{
+    char **argv = sol_argv();
+    int port = 8080, c, opt_idx,  argc = sol_argc();
+    static const struct option opts[] = {
+        { "port", required_argument, NULL, 'p' },
+        { "help", no_argument, NULL, 'h' },
+        { 0, 0, 0, 0 }
+    };
+
+    sol_buffer_init(&value);
+    while ((c = getopt_long(argc, argv, "p:h", opts, &opt_idx)) != -1) {
+        switch (c) {
+        case 'p':
+            port = atoi(optarg);
+            break;
+        case 'h':
+        default:
+            fprintf(stderr,
+                "Usage:\n\t%s [-p <port >]\n", argv[0]);
+            sol_quit_with_code(EXIT_SUCCESS);
+            return;
+        }
+    }
+
+    /* always set stdin to non-block before we use sol_fd_add() on it,
+     * otherwise we may block reading and it would impact the main
+     * loop dispatching other events.
+     */
+    if (sol_util_fd_set_flag(STDIN_FILENO, O_NONBLOCK) < 0) {
+        fprintf(stderr, "ERROR: cannot set stdin to non-block.\n");
+        goto err_watch;
+    }
+
+    stdin_watch = sol_fd_add(STDIN_FILENO,
+        SOL_FD_FLAGS_IN | SOL_FD_FLAGS_HUP | SOL_FD_FLAGS_ERR, on_stdin, NULL);
+    if (!stdin_watch) {
+        fprintf(stderr, "ERROR: Failed to watch stdin\n");
+        goto err_watch;
+    }
+
+    c = sol_buffer_set_slice(&value,
+        sol_str_slice_from_str("data: Soletta string server, set the value using the keyboard\n\n"));
+    if (c < 0) {
+        fprintf(stderr, "ERROR: Failed to set buffer's value\n");
+        goto err_buffer;
+    }
+
+    server = sol_http_server_new(port);
+    if (!server) {
+        fprintf(stderr, "ERROR: Failed to create the server\n");
+        goto err_server;
+    }
+
+    if (sol_http_server_register_handler(server, "/", request_cb, NULL) < 0) {
+        fprintf(stderr, "ERROR: Failed to register the handler\n");
+        goto err_handler;
+    }
+
+    printf("HTTP server at port %d.\nDefault reply set to '%.*s'\n",
+        port, SOL_STR_SLICE_PRINT(sol_buffer_get_slice(&value)));
+
+    sol_buffer_init(&value);
+    sol_ptr_vector_init(&responses);
+    return;
+
+err_handler:
+err_server:
+    sol_buffer_fini(&value);
+err_buffer:
+    sol_fd_del(stdin_watch);
```

Comment:
Remenber to set stdin_watch to NULL here, otherwise you will double free it in shutdown_server